docs/src/api/class-page.md

```diff
@@ -2024,6 +2024,38 @@ Name of the function on the window object
 
 Callback function which will be called in Playwright's context.
 
+## async method: Page.extract
+* since: v1.58
+* langs: js
+- returns: <[any]>
+
+Extract information from the page using the agentic loop, return it in a given Zod format.
+
+**Usage**
+
+```js
+await page.extract('List of items in the cart', z.object({
+  title: z.string().describe('Item title to extract'),
+  price: z.string().describe('Item price to extract'),
+}).array());
+```
+
+### param: Page.extract.query
+* since: v1.58
+- `query` <[string]>
+
+Task to perform using agentic loop.
+
+### param: Page.extract.schema
```

Comment:
Make it an option and return free text by default?

Reply:
Can't make it an option due to being frequent, so would need to do overload.